csharp/native/SubTasking/Worker/ArmoniK.Samples.SubTasking.Worker/SubTaskingWorker.cs

```diff
@@ -0,0 +1,267 @@
+using ArmoniK.Api.Common.Utils;
+using ArmoniK.Api.gRPC.V1;
+using ArmoniK.Api.gRPC.V1.Agent;
+using ArmoniK.Api.Worker.Worker;
+
+using Google.Protobuf;
+using Google.Protobuf.Collections;
+using Google.Protobuf.WellKnownTypes;
+
+using Microsoft.Extensions.Logging;
+
+using Empty = ArmoniK.Api.gRPC.V1.Empty;
+
+namespace ArmoniK.Samples.SubTasking.Worker
+{
+  public class SubTaskingWorker : WorkerStreamWrapper
+  {
+    /// <summary>
+    ///   Initializes an instance of <see cref="SubTaskingWorker" />
+    /// </summary>
+    /// <param name="loggerFactory">Factory to create loggers</param>
+    /// <param name="provider">gRPC channel provider to send tasks and results to ArmoniK Scheduler</param>
+    public SubTaskingWorker(ILoggerFactory      loggerFactory,
+                            GrpcChannelProvider provider)
+      : base(loggerFactory,
+             provider)
+      => logger_ = loggerFactory.CreateLogger<SubTaskingWorker>();
+
+    public override async Task<Output> Process(ITaskHandler taskHandler)
+    {
+      using var scopedLog = logger_.BeginNamedScope("Execute task",
+                                                    ("sessionId", taskHandler.SessionId),
+                                                    ("taskId", taskHandler.TaskId));
+      try
+      {
+        // We may use TaskOptions.Options to send a field UseCase where we inform
+        // what should be executed
+        var useCase = taskHandler.TaskOptions.Options["UseCase"];
+
+        switch (useCase)
+        {
+          case "Launch":
+            await Launch(taskHandler);
+            break;
+          case "Joiner":
+            await Joiner(taskHandler);
+            break;
+          case "HelloWorker":
+            await HelloWorker(taskHandler);
+            break;
+          default:
+            return new Output
+                   {
+                     Error = new Output.Types.Error
+                             {
+                               Details = "UseCase not found",
+                             },
+                   };
+        }
+      }
+      catch (Exception e)
+      {
+        logger_.LogError(e,
+                         "Error during task computing.");
+        return new Output
+               {
+                 Error = new Output.Types.Error
+                         {
+                           Details = e.Message,
+                         },
+               };
+      }
+
+      return new Output
+             {
+               Ok = new Empty(),
+             };
+    }
+
+    // Launching helloWorkers and joiner, both are tasks created by our primary task
+    public async Task Launch(ITaskHandler taskHandler)
+    {
+      logger_.Log(LogLevel.Debug,
+                  "Launching workers");
+      await SubmitJoiner(taskHandler,
+                         await SubmitWorkers(taskHandler));
+    }
+
+    public async Task<List<string>> SubmitWorkers(ITaskHandler taskHandler)
```

Comment:
As a public function, you should add xml-doc
Either you add the doc or you make them private